### PR TITLE
fix(CI): updates to restore CI lint and integration tests

### PR DESCRIPTION
**Pull request by apeabody**
#### Description
- Updates to restore Integration test (abm-gce-gpu)
  - Switched to currently available GPU type/location, and allocate only to worker nodes
  - With the new dedicated worker node template, also removed nested virtualization from control plane and admin workstation as not required
- Updates to fix and re-enable (migrated from cloudbuild trigger to GH workflow) Lint tests

### CODE REVIEW

File: anthos-bm-gcp-terraform/terraform.tfvars.sample [hunk 1]
```diff
@@ -15,5 +15,5 @@ instance_count                = {
 #nfs_server                   = true
 #gpu                          = {
 #                                 count = 1,
-#                                 type = "nvidia-tesla-k80"
+#                                 type = "nvidia-tesla-t4"
```
Comment (NimJay) [inline anthos-bm-gcp-terraform/terraform.tfvars.sample:18]: Question: Is there a reason for this change (`nvidia-tesla-k80` to `nvidia-tesla-t4`)?

File: anthos-bm-gcp-terraform/variables.tf [hunk 2]
```diff
@@ -41,7 +41,7 @@ variable "region" {
 variable "zone" {
   description = "Zone within the selected Google Cloud Region that is to be used"
   type        = string
-  default     = "us-central1-a"
+  default     = "us-central1-b"
```
Comment (NimJay) [inline anthos-bm-gcp-terraform/variables.tf:44]: Question: What was the reason for this change (`us-central1-a` to `us-central1-b`)?

File: anthos-bm-gcp-terraform/main.tf [hunk 3]
```diff
@@ -108,6 +108,33 @@ module "create_service_accounts" {
...
+  source_image_family  = var.image_family     # --image-family=ubuntu-2004-lts
+  source_image_project = var.image_project    # --image-project=ubuntu-os-cloud
+  machine_type         = var.machine_type     # --machine-type $MACHINE_TYPE
+  disk_size_gb         = var.boot_disk_size   # --boot-disk-size 200G
+  disk_type            = var.boot_disk_type   # --boot-disk-type pd-ssd
+  network              = var.network          # --network default
+  tags                 = var.tags             # --tags http-server,https-server
+  min_cpu_platform     = var.min_cpu_platform # --min-cpu-platform "Intel Haswell"
+  can_ip_forward       = true                 # --can-ip-forward
+  # Disable oslogin explicitly since we rely on metadata based ssh-key (issues/70).
```
Comment (NimJay) [inline anthos-bm-gcp-terraform/main.tf:130]: **Question:** Is the exclusion of `enable_nested_virtualization = var.enable_nested_virtualization` intentional?

File: anthos-bm-gcp-terraform/main.tf [hunk 4]
```diff
@@ -175,7 +202,7 @@ module "worker_vm_hosts" {
   zone              = var.zone
   network           = var.network
   vm_names          = local.worker_vm_names
-  instance_template = module.instance_template.self_link
+  instance_template = module.instance_template_worker.self_link
```
Comment (NimJay) [inline anthos-bm-gcp-terraform/main.tf:205]: **Question:** What was reason for introducing a new instance template (`instance_template_worker`)?

**Review by NimJay — APPROVED** [review]
@apeabody, thank you for being back the `lint` workflow! :)
Overall, the changes look good — especially since all the GitHub checks are passing.
I just left some questions for you about certain code changes. I believe some context (short explanations) would help future maintainers.

Approved. Feel free to merge when you're comfortable. (Let me know if you don't have access.)

File: anthos-bm-gcp-terraform/terraform.tfvars.sample (hunk 1 above)
Comment (apeabody) [reply anthos-bm-gcp-terraform/terraform.tfvars.sample:18]: Yes, the `nvidia-tesla-k80` are EOL and no longer available (blocks the CI build): https://cloud.google.com/compute/docs/eol/k80-eol

File: anthos-bm-gcp-terraform/variables.tf (hunk 2 above)
Comment (apeabody) [reply anthos-bm-gcp-terraform/variables.tf:44]: The `nvidia-tesla-t4` are not available in `us-central1-a`, the recommendation is `us-central1-b` or `us-central1-f`.

File: anthos-bm-gcp-terraform/main.tf (hunk 3 above)
Comment (apeabody) [reply anthos-bm-gcp-terraform/main.tf:130]: Yes - As this template has been re-scoped to just the control plane and admin workstation, there is no need for nested virtualization.  Nested virtualization was added to enable the VM Runtime feature, and is only needed on worker nodes.

**Comment by apeabody** [comment]
> @apeabody, thank you for bringing back the `lint` workflow! :) Overall, the changes look good — especially since all the GitHub checks are passing. I just left some questions for you about certain code changes. I believe some context (short explanations) would help future maintainers.
> 
> Approved. Feel free to merge when you're comfortable. (Let me know if you don't have the ability to merge.)
> 
> **Update:** I gave you admin access to the repo.

Thanks @NimJay! Let me know if you have any questions on the responses.  I'll contact you OOB about merging process for this repo.

**Review by NimJay — APPROVED** [review]
Thanks for answering all my questions. :) Everything makes sense. Feel free to merge whenever.

**Pull request merged**